dbt/include/trino/macros/crossdb/datediff.sql

```diff
@@ -23,6 +23,6 @@
     {% elif datepart == 'millisecond' %}
         (to_milliseconds((CAST({{ second_date }} AS TIMESTAMP) - CAST({{ first_date }} AS TIMESTAMP))))
     {% else %}
-        {{ exceptions.raise_compiler_error("Unsupported datepart for macro datediff in Trino: {!r}".format(datepart)) }}
+        {% if execute %}{{ exceptions.raise_compiler_error("Unsupported datepart for macro datediff in Trino: {!r}".format(datepart)) }}{% endif %}
```

Comment:
what problem were you trying to solve?

Reply:
I found in the meantime: 
https://github.com/dbt-labs/dbt-core/blob/a7ff003d4f74763f126e77a0fc9c55e7a3311b3d/core/dbt/context/providers.py#L914-L970